src/components/select-custom/select-custom.component.ts

```diff
@@ -0,0 +1,128 @@
+import { ChangeDetectionStrategy, Component, ContentChild, ElementRef, EventEmitter, forwardRef, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges, TemplateRef, ViewChild } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'ux-select-custom',
+    templateUrl: './select-custom.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => SelectCustomComponent)
+        }
+    ],
+    host: {
+        '[style.width.px]': 'width'
+    }
+})
+export class SelectCustomComponent<T> implements ControlValueAccessor, OnChanges, OnInit, OnDestroy {
+
+    /** Define the selected item */
+    @Input() selected: T;
+
+    /** Hide the filter input */
+    @Input() hideFilter: boolean;
+
+    /** Debounce the filter change event emitter */
+    @Input() filterChangeDebounce: number = 500;
+
+    /** Define the width of the component */
+    @Input() width: number;
```

Comment:
The button and the drop-down have to have the same width and I am not sure how to achieve that without explicitly setting the width on both components. If you have an idea, please let me know.

Reply:
You could add a reference to the button and then set the width of the dropdown based on it, eg:

```html
<button #button>

<ux-menu [style.width.px]="button.offsetWidth">
```

Or something along similar to that

Reply:
Check out how the `ux-select` component does it. Both input and dropdown are children of a common parent container and have `width: 100%`. The key to this is setting `position: relative` on the parent, so that it counts as the nearest positioned ancestor of both children.

![image](https://user-images.githubusercontent.com/26114101/62775780-8cb95280-baa0-11e9-8f51-f91558fcff7a.png)

Reply:
Thank you for your suggestions! I implemented the first one, the second I tried yesterday - however, ux-menu does not put button and drop-down in the same container.